src/core/ComposableElement.php

```diff
@@ -41,7 +41,19 @@ protected function init(): void {
     foreach ($children as $child) {
       $this->appendChild($child);
     }
-    $this->setAttributes($attributes);
+
+    foreach ($attributes as $key => $value) {
+      if (self::isSpreadKey($key)) {
+        invariant(
+          $value instanceof :x:composable-element,
+          "Only XHP can be used with an attribute spread operator",
```

Comment:
This is for typechecker refinement (i.e. the runtime/typechecker don't support non-XHP spreading anyways) rather than a "product" decision to not support things like containers, right?

Reply:
I think it only makes sense to be able to spread shapes (so that the typechecker can have a handle on what's going on), not any ol' container.

The most common use case for attribute spread, at least in my experience so far, is:
```
return <foo {...$this} another-attr="bar" />"
```

which is well-served even with the restriction of "only XHP". I would not be opposed to implementing it for shapes if we can find convincing usecases.